pkg/ruler/rule_backup_manager.go

```diff
@@ -0,0 +1,156 @@
+package ruler
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/prometheus/model/rulefmt"
+	"github.com/prometheus/prometheus/promql/parser"
+	promRules "github.com/prometheus/prometheus/rules"
+
+	"github.com/cortexproject/cortex/pkg/ruler/rulespb"
+)
+
+// Implements GroupLoader interface but instead of reading from a file when Load is called, it returns the
+// rulefmt.RuleGroup it has stored
+type loader struct {
+	ruleGroups map[string][]rulefmt.RuleGroup
+}
+
+func (r *loader) Load(identifier string) (*rulefmt.RuleGroups, []error) {
+	return &rulefmt.RuleGroups{
+		Groups: r.ruleGroups[identifier],
+	}, nil
+}
+
+func (r *loader) Parse(query string) (parser.Expr, error) {
+	return parser.ParseExpr(query)
+}
+
+// rulesBackupManager is an in-memory store that holds []promRules.Group of multiple users. It only stores the Groups,
+// it doesn't evaluate them.
+type rulesBackupManager struct {
```

Comment:
I have renamed the method `backUpRuleGroups` to `setRuleGroups` because the name of the method to get the data is called `getRuleGroups`. I would like to keep the name of the package as is because `storage` or `store` is generally used to refer to persistent/remote storage in the codebase.